engine/src/sysunxdate.cpp

```diff
@@ -171,6 +171,34 @@ static MCStringRef string_prepend(MCStringRef t_string, unichar_t t_char)
 	return t_result;
 }
 
+
+// PM-2015-09-07: [[ Bug 9942 ]] On Linux, the short/abbr system time should
+//  not return the seconds
+static MCStringRef remove_seconds(MCStringRef p_input)
+{
+    MCStringRef t_new_string;
+    MCRange t_range;
+
+    if (MCStringFind(p_input, MCRangeMake(0, UINDEX_MAX), MCSTR(":%S"), kMCStringOptionCompareExact, &t_range))
+    {
+        // If :%S is found, then we remove it
+        MCAutoStringRef t_mutable_copy;
+        /* UNCHECKED */ MCStringMutableCopy(p_input, &t_mutable_copy);
+
+        /* UNCHECKED */ MCStringRemove(*t_mutable_copy, t_range);
+
+        /* UNCHECKED */ MCStringCopy(*t_mutable_copy, t_new_string);
+    }
+    else
+    {
+        // Otherwise we take the whole string
+        t_new_string = MCValueRetain(p_input);
+    }
+	
+	return t_new_string;
+}
+
```

Comment:
This would be more efficient if you did 

```
	MCStringRef t_mutable_copy = nil;
	/* UNCHECKED */  MCStringMutableCopy(p_string, t_mutable_copy)
	/* UNCHECKED */  MCStringRemove(*t_mutable_copy, t_range)
        /* UNCHECKED */  MCStringCopyAndRelease(*t_mutable_copy, t_new_string)
```

that way the mutable string just gives its buffer to t_new_string.

Not that the input string is ever going to be particularly long, but still :smile: